### DISCUSSION TITLE
beginners need help ;-)

### DISCUSSION THREAD

**Discussion by kleinrene** [bpatrik/pigallery2#307] [Q&A]
I have a Raspberry Pi 4 Model B (4GB)

Installation via Raspberry Pi Imager 1.61
Raspberry Pi OS Lite
Release date: March 4th 2021
Kernel version: 5.10
Size: 442MB

Store empty file with the designation "ssh" in the file system

Start Raspberry and access via putty

$ sudo apt-get update && sudo apt-get upgrade

Then follow the instructions (https://github.com/bpatrik/pigallery2/blob/master/docker/README.md) and end up at:

sudo pip install docker-compose -> pyrsistent requires Python '>=3.5' but the running Python is 2.7.16

The installation and deinstallation of python has not been successful so far and it seems that I cannot see the forest for the trees.

**Comment by bpatrik** [comment]
Looking at my notes, Last time (8 months ago) i did this whit my RPI4 to make it work:
Does this help?
Please take it a grain of salt, havent double checked them now, something might be missing

-----------------------------------------------

install docker https://github.com/bpatrik/pigallery2/blob/master/docker/README.md
```
curl -sSL https://get.docker.com | sh
```

compose:
```
sudo apt-get install libffi-dev libssl-dev
sudo apt-get install -y python python-pip
sudo apt-get remove python-configparser
sudo pip install docker-compose
```

Get pigallery files
```
wget https://raw.githubusercontent.com/bpatrik/pigallery2/master/docker/docker-compose/default/docker-compose.yml
wget https://raw.githubusercontent.com/bpatrik/pigallery2/master/docker/docker-compose/default/nginx.conf
```


install certbot for ssl (https)
```
sudo apt-get install certbot python-certbot-nginx
sudo certbot --nginx # this way we have a proper nginx settings
sytemctl disable nginx #the hots's nginx binds to port 80/443 that prevents starting dockerised nginx nad cerbot's ssh check
```

**Reply by bpatrik** [reply]
Just saw that you are using `Raspberry Pi OS Lite`. I'm using raspbian, so probably my comment does not apply.

**Reply by kleinrene** [reply]
sytemctl -> systemctl?

**Comment by kleinrene** [comment]
I came a little further and was able to find the fault.

manual: sudo pip install docker-compose
other manuel: sudo pip3 -v install docker-compose (https://dev.to/rohansawant/installing-docker-and-docker-compose-on-the-raspberry-pi-in-5-simple-steps-3mgl)

Now the next challenge, the installation of the rest of the manual was actually error-free ...

pigallery2 is up-to-date
nginx is up-to-date

but the page is unfortunately not available ...

**Reply by bpatrik** [reply]
Check logs:
```
docker ps -a
docker logs <id>
```
And make sure that you have updated both docker compose file and nginx.conf  (#CHANGE ME parts)

**Reply by kleinrene** [reply]
5/22/2021, 4:32:09 PM[INFO] [server] running diagnostics...
(node:1) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
5/22/2021, 4:32:10 PM[INFO] [JobManager] Running job schedules
5/22/2021, 4:32:10 PM[INFO] [server] Listening on port 80
(node:20) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)

I think I should change the operating system, are you using Raspberry Pi OS with desktop?

**Reply by bpatrik** [reply]
You do not need the desktop on the server. I also do not use it. 

Raspberry logs look good. I think your issue is with nginx or port binding to the host.

**Reply by bpatrik** [reply]
you can try:
`sudo netstat -tulpn | grep LISTEN`

That should list what is listening where. 

It should contain:
```
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      1053/docker-proxy
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      1016/docker-proxy
tcp6       0      0 :::80                   :::*                    LISTEN      1061/docker-proxy
tcp6       0      0 :::443                  :::*                    LISTEN      1039/docker-proxy
```

if you are using nginx within docker too.

**Comment by kleinrene** [comment]
I gave up the fight with the Raspberry for the time being. I am now running the application on a QNAP with an upstream reverse proxy (Nginx Proxy Manager).
